react/week3/extending-todo-app/src/App.js

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from 'react';
+import List from './List';
+import './App.css';
+
+function App() {
+  const [description,setDescription]=useState("");
+  const [date,setDate]=useState("");
+  const [list, setList] = useState([]);
+
+  useEffect(() => {
+    fetch("https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw")
+      .then(res => res.json())
+      .then(
+        (result) => {
+          setList(result);
+        },
+        (error) => {
+          console.log(error)
+        }
+      )
+  }, [])
+
+
+
+  function editListItem(id,newDescription){
+    const newList=[...list];
+    const objIndex = newList.findIndex((obj => obj.id === id));
+    newList[objIndex].description=newDescription;
+    setList(newList);
+  }
+  
+ 
+  function deleteListItem(id){
+    const newList=[...list];
+    const index=newList.map(x=>{
+      return x.id;
+    }).indexOf(id);
+    newList.splice(index,1);
+    setList(newList);
+  }
+ function addNewTodo(newDescription,newDate){
+  const dateAsString=newDate.toString();
+  const newTodo={
+
+    id:list.length+1,
```

Comment:
_Sunil had the same solution here, so I just copied my comment from his PR:_ 

This is not the best way to construct a unique ID.
Let me show you an example:
You have an array of todos ---> `[todo1, todo2, todo3]`
Let's say the user adds a new todo `todo4`, `todo4` will have the id of todos.length + 1 = 4.
Then the user realizes they don't need `todo1` so they remove that from the list.
Now you have `[todo2, todo3, todo4]`
If the user adds a new todo, `todo5`, the id will be todos.length + 1 = 4 again. So you end up with 2 items that have the same id.

Here you can find some unique ID alternatives:
https://dev.to/rahmanfadhil/how-to-generate-unique-id-in-javascript-1b13